### ISSUE TITLE
[Bug]: child_process functions not working in packaged app

### ISSUE THREAD

**Issue by ttrepka** [electron/electron#33070]
### Preflight Checklist

- [X] I have read the [Contributing Guidelines](https://github.com/electron/electron/blob/main/CONTRIBUTING.md) for this project.
- [X] I agree to follow the [Code of Conduct](https://github.com/electron/electron/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/electron/electron/issues) for a bug report that matches the one I want to file, without success.

### Electron Version

17.0.0

### What operating system are you using?

macOS

### Operating System Version

macOS Mojave 10.14.6

### What arch are you using?

x64

### Last Known Working Electron version

16.10.0

### Expected Behavior

I can run `child_process.exec`, `execFile` and `spawn` from the main process in a packaged app.

### Actual Behavior

`child_process.exec`, `execFile` and `spawn` work when I "Run" the project in Electron Fiddle but not when I package using "Tasks" > "Package fiddle...".

### Testcase Gist URL

https://gist.github.com/ttrepka/701306ff2283166644e8f76efec3ad81

**Comment by sergiou87** [comment]
Not sure if it's the same but definitely sounds similar: when trying to run `spawn` from the main process in a packaged Fiddle moved to `/Applications` on macOS Catalina, I'm getting a crash. But then the app works normally. I've used this Fiddle to reproduce the issue: https://gist.github.com/cf33b295daf2b42fd3f50a3bec6aea00

It crashes on all 17.x versions and 18.0.0-alpha, but not on 16.x.

On GitHub Desktop, we're getting reports of these crashes since we upgraded the app to 17.0.1: https://github.com/desktop/desktop/issues/13974

**Comment by sergiou87** [comment]
Oh, not sure about this, but I think my issue is better described in https://github.com/electron/electron/issues/327

**Comment by ckerr** [comment]
I _think_ @sergiou87 is right, this does look like a duplicate of the other issue. @ttrepka I see a fix for the other issue has been merged -- could you try this fix out with a version of Electron [as described here](https://github.com/desktop/desktop/issues/13974#issuecomment-1057062961) and see if the problem is fixed or if it persists?

**Comment by ttrepka** [comment]
@ckerr I don't know how to test the fix of #32718, I guess I need to wait for the next release of 17.x.y?
As for the fix in Github Desktop you mentioned, it seems they only moved all calls to `spawn` to the renderer process, which is not a fix, just a workaround.

**Comment by ape-casear** [comment]
same problem
```
childProcess = cp.fork(path.join(__dirname, './child.js'))

    childProcess.on('error', (error) => {
        log('[CHILD] error:' + JSON.stringify(error))
    })
    childProcess.on('message', (msg) => {
        log('[CHILD] message' + msg)
    })
    childProcess.on('disconnect', () => {
        log('[CHILD] disconnect')
    })
    childProcess.on('close', () => {
        log('[CHILD] close')
    })
    childProcess.on('spawn', () => {
        log('[CHILD] spawn')
    })
    childProcess.on('exit', (code, sign) => {
        log(`[CHILD] exit code:${code}  sign:${sign}`)
    })

    log(`[CHILD] pid:${childProcess.pid}, connected: ${childProcess.connected}`)
```
expect:
[CHILD] pid:341, connected: true
[CHILD] spawn
[CHILD] message I am child
[CHILD] disconnect
[CHILD] exit code:0  sign:null
[CHILD] close

actually:
[CHILD] pid:650, connected: true
[CHILD] spawn
[CHILD] disconnect
[CHILD] exit code:null  sign:SIGSEGV
[CHILD] close

**Comment by ttrepka** [comment]
I guess this really was a duplicate of https://github.com/electron/electron/issues/32718, fixed by https://github.com/electron/electron/pull/33114, released in Electron 17.1.1.
I'm closing this.
